src/name/bizna/ocarmsim/ROMRegion.java

```diff
@@ -13,8 +12,8 @@
 	// cribbed from JARMArchitecture
 	private static final int ROM_MIN_BIT_DEPTH = 12; // 4K ROM
 	
-	byte[] array;
-	int arrayMask;
+	private final byte[] array;
+	private final int arrayMask;
```

Comment:
It's a little frightening how, even though I _know_ HotSpot optimizes accessors to be just as fast, there were still so many places where my brain _insisted_ that there should be direct access to fields...